src/NLog/Config/XmlLoggingConfiguration.cs

```diff
@@ -272,7 +272,7 @@ public override IEnumerable<string> FileNamesToWatch
         public override LoggingConfiguration Reload()
         {
             if (!string.IsNullOrEmpty(_originalFileName))
-                return new XmlLoggingConfiguration(_originalFileName, LogFactory);
+                return LogFactory.CreateConfig(_originalFileName); //TODO NLog 5: should call LoadConfiguration which set LogFactory.Configuration
```

Comment:
Still curious why you need to modify this method? Why not just have a method called `LogFactory.ReloadConfiguration()` instead of the config assigning itself?

Reply:
because `new XmlLoggingConfiguration(_originalFileName, LogFactory);` doens't use the _configLoader, and thus you well get this.

```c#
   var appEnv = new Mocks.AppEnvironmentMock( ...)
  var configLoader = new LoggingConfigurationFileLoader(appEnv);
  var logFactory = new LogFactory(configLoader);

  // Act
  logFactory.Setup().
  LoadConfigurationFromFile().
  // config loaded from configLoader 

  // crash:  System.IO.FileNotFoundException - Could not find file 'D:\nlog\NLog\tests\NLog.UnitTests\bin\Debug\net452\NLog.config'.
  logFactory.Configuration = logFactory.Configuration.Reload()
```

Reply:
Would it not be better to change `XmlLoggingConfiguration` to use the LogFactory-AppEnvironment for loading the XmlConfig-file ? (Improve mocking all around, instead of just Reload)

Reply:
Good point

Reply:
Have created #3941 that improves the mocking of `XmlReader.Create` for `XmlLoggingConfiguration`.

Reply:
> Why not in the future just have a method called LogFactory.ReloadConfiguration() instead of the config assigning itself?

I think also this is better! I think using the setter was a mistake, also I really need to use the reload event, not the changed. So fixed that